ecc/goldilocks/point.go

```diff
@@ -169,3 +106,48 @@ func (P *Point) Add(Q *Point) {
 	fp.Mul(x3, E, F)      // X = E * F
 	fp.Mul(y3, G, H)      // Y = G * H, T = E * H
 }
+
+// UnmarshalBinary if succeeds constructs a point by decoding the first
+// CurveEncodingSize bytes of in.
+func (P *Point) UnmarshalBinary(in []byte) error {
+	if len(in) < CurveEncodingSize {
+		return errInvalidDecoding
+	}
+	signX := in[fp.Size] >> 7
+	copy(P.y[:], in[:fp.Size])
+	p := fp.P()
+	if !isLessThan(P.y[:], p[:]) {
+		return errInvalidDecoding
+	}
+
+	u, v := &fp.Elt{}, &fp.Elt{}
+	one := fp.One()
+	fp.Sqr(u, &P.y)                // u = y^2
+	fp.Mul(v, u, &paramD)          // v = dy^2
+	fp.Sub(u, u, &one)             // u = y^2-1
+	fp.Sub(v, v, &one)             // v = dy^2-1
+	isQR := fp.InvSqrt(&P.x, u, v) // x = sqrt(u/v)
+	if !isQR {
```

Comment:
This is not constant time.

Reply:
updated take a look

Reply:
Looks good.

Reply:
It might be hard though to write constant time algorithms using this function as it returns an error instead of an int.  In [go-ristretto](https://github.com/bwesterb/go-ristretto) I have two variants: `Equals` and `EqualsI`.  The former returns the idiomatic `bool` and the latter an `int`, which allows one to write constant time code.